drivers/include/sdmmc/sdmmc.h

```diff
@@ -0,0 +1,1632 @@
+ */
+typedef enum {
+    SDMMC_CSD_V1 = 0,   /**< CSD Version 1.0 (SDSC and MMC) */
+    SDMMC_CSD_V2 = 1,   /**< CSD Version 2.0 (SDHC/SDXC) or Version 1.1 (MMC) */
+    SDMMC_CSD_V3 = 2,   /**< CSD Version 3.0 (SDUC) or Version 1.2 (MMC) */
+    SDMMC_CSD_Vx = 3,   /**< Reserved (SD) or Version in EXT_CSD (MMC) */
+} sdmmc_csd_version_t;
+
+/** CSD register size in byte (CSD is 128 bit) */
+#define SDMMC_CSD_REG_SIZE  (16)
+
+/**
+ * @brief   CSD register structure Version 1.0
+ *
+ * @see Physical Layer Simplified Specification Version 9.00, Section 5.3.2,
+ *      Table 5.4 [[sdcard.org](https://www.sdcard.org)]
+ */
+typedef struct {
+    uint8_t CSD_STRUCTURE:2;        /**< CSD structure [127:126] */
+    uint8_t TAAC:8;                 /**< data read access-time-1 [119:112] */
+    uint8_t NSAC:8;                 /**< data read access-time-2 in CLK cycles [111:104] */
+    uint8_t TRAN_SPEED:8;           /**< max. data transfer rate [103:96] */
+    uint16_t CCC:12;                /**< card command classes [95:84] */
+    uint8_t READ_BL_LEN:4;          /**< max. read data block length [83:80] */
+    uint8_t READ_BL_PARTIAL:1;      /**< partial blocks for read allowed [79] */
+    uint8_t WRITE_BLK_MISALIGN:1;   /**< write block misalignment [78] */
+    uint8_t READ_BLK_MISALIGN:1;    /**< read block misalignment [77] */
+    uint8_t DSR_IMP:1;              /**< DSR implemented [76] */
+    uint16_t C_SIZE:12;             /**< device size [73:62] */
+    uint8_t VDD_R_CURR_MIN:3;       /**< max. read current VDD min [61:59] */
+    uint8_t VDD_R_CURR_MAX:3;       /**< max. read current VDD max [58:56] */
+    uint8_t VDD_W_CURR_MIN:3;       /**< max. write current VDD min [55:53] */
+    uint8_t VDD_W_CURR_MAX:3;       /**< max. write current VDD max [52:50] */
+    uint8_t C_SIZE_MULT:3;          /**< device size multiplier [49:47] */
+    uint8_t ERASE_BLK_EN:1;         /**< erase single block enable [46] */
+    uint8_t SECTOR_SIZE:7;          /**< erase sector size [45:39] */
+    uint8_t WP_GRP_SIZE:7;          /**< write protect group size [38:32] */
+    uint8_t WP_GRP_ENABLE:1;        /**< write protect group enable [31] */
+    uint8_t R2W_FACTOR:3;           /**< write speed factor [28:26] */
+    uint8_t WRITE_BL_LEN:4;         /**< max. write data block length [25:22] */
+    uint8_t WRITE_BL_PARTIAL:1;     /**< partial blocks for write allowed [21] */
+    uint8_t FILE_FORMAT_GRP:1;      /**< File format group [15] */
+    uint8_t COPY:1;                 /**< copy flag [14] */
+    uint8_t PERM_WRITE_PROTECT:1;   /**< permanent write protection [13] */
+    uint8_t TMP_WRITE_PROTECT:1;    /**< temporary write protection [12] */
+    uint8_t FILE_FORMAT:2;          /**< File format [11:10] */
+    uint8_t WP_UPC:1;               /**< write protection until power cycle [9] */
+    uint8_t CSD_CRC:8;              /**< CRC including End bit 1b [7:0] */
+} sdmmc_csd_v1_t;
```

Comment:
```suggestion
}  __attribute__((packed)) sdmmc_csd_v1_t;
```

same here and below since those are read directly from memory